lightning-data-persister/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use lightning::ln::data_persister::ChannelDataPersister;
```

Comment:
I'm not sure what the performance story is for this module, but syncing to disk can sometimes take 100s of milliseconds. If this isn't good enough, it may be worth exploring a model where writes are queued to a separate writer thread to avoid any contention.

Reply:
That does sound like better performance. I'd probably file an issue + do it as a follow-up PR though, unless it seems crucial for the v1 of this.

Reply:
Definitely not important for v1, just want to point it out as a good place to look for perf issues if they surface in the future. Writing queues also introduce a class of bugs and failures that are harder to test and focusing on getting something simple that works first is perfect.

Reply:
Sounds good, I ended up adding a comment about this in commit `f add comment about future queue-based system`